crates/control/src/calibration_controller.rs

```diff
@@ -0,0 +1,299 @@
+    >,
+    field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
+    parameters: Parameter<CameraCalibrationControllerParameters, "calibration_controller">,
+    calibration_measurements: AdditionalOutput<Vec<Measurement>, "calibration_measurements">,
+    last_calibration_corrections:
+        AdditionalOutput<Option<Corrections>, "last_calibration_corrections">,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub calibration_command: MainOutput<CalibrationCommand>,
+}
+
+impl CalibrationController {
+    pub fn new(_context: CreationContext) -> Result<Self> {
+        Ok(Self {
+            current_calibration_command: CalibrationCommand::default(),
+            look_at_list: generate_look_at_list().unwrap(),
+            current_look_at_index: 0,
+            current_measurements: vec![],
+            current_primary_phase_is_calibration: false,
+            last_calibration_corrections: None,
+        })
+    }
+
+    pub fn cycle(&mut self, mut context: CycleContext) -> Result<MainOutputs> {
+        let calibration_grame_state_active =
+            matches!(context.primary_state, PrimaryState::Calibration);
+        if !calibration_grame_state_active {
+            self.current_calibration_command = CalibrationCommand::INACTIVE;
+            return Ok(MainOutputs::default());
+        }
+
+        let primary_state_transitioned_to_calibration =
+            calibration_grame_state_active && !self.current_primary_phase_is_calibration;
+        self.current_primary_phase_is_calibration = calibration_grame_state_active;
+
+        if primary_state_transitioned_to_calibration {
+            self.current_measurements = vec![];
+        }
+
+        let current_cycle_time = context.cycle_time;
+        let changed_command: Option<CalibrationCommand> = self.get_next_command(
+            primary_state_transitioned_to_calibration,
+            current_cycle_time,
+            &context,
+        );
+
+        if let Some(new_phase) = changed_command {
+            info!("Phase change detected: {:?}", new_phase);
+            self.current_calibration_command = new_phase;
+        }
+
+        context
+            .calibration_measurements
+            .fill_if_subscribed(|| self.current_measurements.clone());
+        context
+            .last_calibration_corrections
+            .fill_if_subscribed(|| self.last_calibration_corrections);
+        Ok(MainOutputs {
+            calibration_command: self.current_calibration_command.clone().into(),
+        })
+    }
+
+    fn get_next_command(
+        &mut self,
+        primary_state_transition_to_calibration: bool,
+        current_cycle_time: &CycleTime,
+        context: &CycleContext,
+    ) -> Option<CalibrationCommand> {
+        let look_at_dispatch_waiting = context.parameters.look_at_dispatch_wait_duration;
+        let initial_stabilization_delay = context
+            .parameters
+            .initial_to_calibration_stabilization_delay;
```

Comment:
Why are these two different names?
Also, do you really need this binding? I would suggest just using `context.stabilization_delay` in place of this.

Reply:
Renamed and removed the bindings.